components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabsToolbarFeature.kt

```diff
@@ -49,50 +50,52 @@ class CustomTabsToolbarFeature(
     internal var readableColor = Color.WHITE
 
     override fun start() {
-        if (initialized) {
-            return
-        }
-        initialized = sessionManager.runWithSession(sessionId) {
-            it.register(sessionObserver)
-            initialize(it)
+        if (!initialized) {
+            initialized = sessionManager.runWithSession(sessionId) { session ->
+                session.register(sessionObserver)
+                session.customTabConfig?.let { config ->
+                    initialize(session, config)
+                    true
+                } ?: false
+            }
         }
     }
 
+    /**
+     * Initializes the custom tab toolbar based on the session's [CustomTabConfig].
+     *
+     * Returns true if initialization succeeded.
+     */
     @VisibleForTesting
-    internal fun initialize(session: Session): Boolean {
-        session.customTabConfig?.let { config ->
-            // Don't allow clickable toolbar so a custom tab can't switch to edit mode.
-            toolbar.onUrlClicked = { false }
-
-            // If it's available, hold on to the readable colour for other assets.
-            config.toolbarColor?.let {
-                readableColor = getReadableTextColor(it)
-            }
+    open fun initialize(session: Session, config: CustomTabConfig) {
```

Comment:
If we are going to go with this approach could we add this annotation [@CallSuper](https://developer.android.com/reference/androidx/annotation/CallSuper.html). As consumers could forget calling super after overridden, as their main purpose could be just changing one thing and they could end up  loss the parent implementation.